xl-v1.go

```diff
@@ -94,6 +136,14 @@ func newXL(disks ...string) (StorageAPI, error) {
 	// Save all the initialized storage disks.
 	xl.storageDisks = storageDisks
 
+	xl.nameSpaceLockMap = make(map[nameSpaceParam]nameSpaceLock)
+	xl.nameSpaceLockMapMutex = &sync.Mutex{}
+	xl.readQuorum = len(xl.storageDisks) / 2
+	xl.writeQuorum = xl.readQuorum + 3
+	if xl.writeQuorum > len(xl.storageDisks) {
+		xl.writeQuorum = len(xl.storageDisks)
+	}
```

Comment:
Reviews can wait keep fixing in the branch. This patch doesnt handle error conditions yet